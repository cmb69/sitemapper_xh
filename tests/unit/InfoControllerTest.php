<?php

/**
 * Copyright 2025 Christoph M. Becker
 *
 * This file is part of Sitemapper_XH.
 *
 * Sitemapper_XH is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Sitemapper_XH is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Sitemapper_XH.  If not, see <http://www.gnu.org/licenses/>.
 */

namespace Sitemapper;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Plib\View;

class InfoControllerTest extends TestCase
{
    public function testExecute(): void
    {
        $model = $this->createStub(Model::class);
        $model->method("installedLanguages")->willReturn(["en", "de"]);
        $view = new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["sitemapper"]);
        $sut = new InfoController(
            "/",
            "en",
            "./plugins/sitemapper",
            "CMSimple_XH 1.8",
            $model,
            $view
        );
        Approvals::verifyHtml($sut->execute());
    }
}
