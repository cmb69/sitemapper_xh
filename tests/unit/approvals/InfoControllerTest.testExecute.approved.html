<div>
  <h1>Sitemapper 3.0-dev</h1>
  <h2>Available Sitemaps</h2>
  <ul>
    <li>
      <a href="/?sitemapper_index" target="_blank">index</a>
    </li>
    <li>
      <a href="/?sitemapper_sitemap" target="_blank">en</a>
    </li>
    <li>
      <a href="/de/?sitemapper_sitemap" target="_blank">de</a>
    </li>
  </ul>
  <h2>System check</h2>
  <p class="xh_success">PHP version ≥ 7.0.0</p>
  <p class="xh_success">CMSimple_XH version ≥ 1.7.0</p>
  <p class="xh_warn">'./plugins/sitemapper/config/' writable</p>
  <p class="xh_warn">'./plugins/sitemapper/languages/' writable</p>
</div>
